index.js

```diff
@@ -124,6 +133,14 @@ function LifxLanPlatform(log, config, api) {
 
             return;
         }
+        else if (this.enabledDevices.indexOf(bulb.id) === -1) {
```

Comment:
along with my requested change at line 108, I think this line needs to be
```else if (this.enabledDevices && this.enabledDevices.indexOf(bulb.id) === -1) {```

otherwise, `this.enabledDevices` will always be an empty array when not defined in the `config.json` (ie: current operation) and all devices will be removed.